src/components/Asset/AssetActions/Download.tsx

```diff
@@ -55,19 +59,35 @@ export default function Download({
     async function init() {
       if (
         asset?.accessDetails?.addressOrId === ZERO_ADDRESS ||
-        asset?.accessDetails?.type === 'free'
+        asset?.accessDetails?.type === 'free' ||
+        (!poolData && asset?.accessDetails?.type === 'dynamic')
       )
         return
       setIsLoading(true)
       setStatusText('Calculating price including fees.')
-      const orderPriceAndFees = await getOrderPriceAndFees(asset, ZERO_ADDRESS)
+
+      const params = {
+        tokenInLiquidity: poolData?.baseTokenLiquidity,
+        tokenOutLiqudity: poolData?.datatokenLiquidity,
+        tokenOutAmount: '1',
+        opcFee: getOpcForToken(poolData.baseToken.address, asset?.chainId),
+        lpSwapFee: poolData?.liquidityProviderSwapFee,
+        publishMarketSwapFee: poolData?.publishMarketSwapFee,
+        consumeMarketSwapFee: '0'
+      } as CalcInGivenOutParams
```

Comment:
this converts from one type to the other so errors in above object creation would not be checked. To get that, we want the variable to be assigned a type, so `const params: CalcInGivenOutParams`